otumat/watch.py

```diff
@@ -5,12 +5,17 @@
 
 
 class OnMyWatch:
-    def __init__(self, watch_file, watch_interval, watch_script, watch_args):
+    def __init__(self, watch_file, watch_interval, watch_init, watch_script, watch_args):
         self.observer = PollingObserver(timeout=watch_interval)
         self.watch_directory = watch_file
         self.watch_script = watch_script
         self.watch_args = watch_args
 
+        if watch_init:
+            self.watch_args = subprocess.Popen(
+                [self.watch_script, *self.watch_args],
+                stdout=subprocess.PIPE).communicate()[0].decode('utf-8').split('\n')[:-1]
+
```

Comment:
Hmm, I'm not sure if this is in the right place. It may be better to put this within `run` as we normally preserve state in the `watch_args` associated with the `Handler`.